src/components/Widget/components/Conversation/components/Messages/components/Carousel/index.js

```diff
@@ -53,6 +53,7 @@ const Carousel = (props) => {
   };
 
   const { linkTarget } = props;
+  const defaultLinkTarget = '_self';  // instead of '_blank'
```

Comment:
Is there a possibility where we render any link which to be opened in a new tab? or is it always be course cards?

Reply:
The answer requires some context:

A comment in [this thread](https://github.com/botfront/rasa-webchat/pull/231) suggests that it's possible to pass some additional metadata in any chatbot reply sent from the server to set whether to open links in a new or in the same tab. Nevertheless, it is not precisely specified how to format the response. I did several attempts to include this in the server responses, as it can be see from this list of commits, so that we didn't have to update this library, but none worked. Therefore, I assumed they meant to customize not the metadata of the message itself but of the Socket.IO response metadata, which is possible to do but probably not worth the effort (we should customize the output channel for our customized Socket.IO implementation on Rasa's back end). I instead thought it was simpler to change the default value for course cards in this library so that when no explicit link target is passed, like it has always happened with our chatbot, the default behavior is to open the course of the corresponding card in the same tab instead of in the new one. I could do the same for other types of "clickable" elements rendered by the library, but course cards are the only ones actually made clickable (with some behavior attached) in our use case. Other elements, such as text messages and buttons, are never supposed to contain links in our use case. We could change their default behavior as well but it would be meaningless as for now. We sometimes embed links inside text messages instead, but that is a whole different story because, in such cases, we simply render text elements that contain links in Markdown, so this library is not responsible for the behavior of those Markdown fragments, which is instead passed from the server depending on the needed Markdown behavior. In these latter cases, links are always opened in new tabs, as asked by the design team. The Jira ticket indeed requires internal links to be opened in the tab page and external ones in a new tab, and at the moment we only have course cards containing internal links.
![immagine](https://github.com/WHOAcademy/rasa-webchat/assets/59971270/773e5f6c-202b-4b6c-8cfe-1baf8e88d1ae)
![immagine](https://github.com/WHOAcademy/rasa-webchat/assets/59971270/e89eaefe-9c7b-4b70-b23f-91481f1ae657)

Reply:
Okay, since I see we limit the carousel to display on the course cards, it is fine, but when we extend it, we may need to rethink this behavior.